llmfoundry/utils/logging_utils.py

```diff
@@ -2,6 +2,29 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import logging
+from typing import List, Union
+
+from composer.loggers import MosaicMLLogger
+from composer.loggers.logger_destination import LoggerDestination
+
+
+def find_mosaicml_logger(
+        loggers: List[LoggerDestination]) -> Union[MosaicMLLogger, None]:
+    return next(
+        (logger for logger in loggers if isinstance(logger, MosaicMLLogger)),
+        None)
+
+
+def get_cloud_provider_from_path(path: str) -> str:
+    """Gets the cloud provider from the a given string path.
+
+    If we see a ':', we know that the service provider is present in the URI.
+    Otherwise, we assume that the model is local.
```

Comment:
> https://github.com/mosaicml/composer/blob/dev/composer/utils/file_helpers.py#L327

gotta love copilot 🤠

Reply:
@irenedea @dakinggg -- can we just grab the providers from `save_folder` / `load_path` in composer? it looks like the same paths from foundry are passed into there, but I want to make sure this is correct.

In this case, I think it may be better to log `save_folder` / `load_path` from composer, rather than foundry. That way we can make sure to grab these providers for non-foundry runs as well. How do y'all feel about this?

Reply:
Yeah those can be grabbed in composer

Reply:
Yes, I think that's fine to log save_folder and load_path from composer.